tests/resources/library_business_logic_suite_related_references.json

```diff
@@ -0,0 +1,44 @@
+{
+  "$schema": "https://raw.githubusercontent.com/OpenAssetIO/OpenAssetIO/main/resources/examples/manager/BasicAssetLibrary/schema.json",
+  "entities": {
+    "entity/original": {
+      "versions": [{"traits": {}}],
+      "relations": [
+        {
+          "traits": {"proxy": {}},
+          "entities": ["entity/proxy/1", "entity/proxy/2"]
+        },
+        {
+          "traits": {"proxy": {"type": "alt"}},
+          "entities": ["entity/proxy/3"]
+        },
+        {
+          "traits": {"source": {}},
+          "entities": ["entity/source"]
+        },
+        {
+          "traits": {"missing": {}},
```

Comment:
[consider] Looks like this case isn't tested? From reading the code it looks like the behaviour for querying the `"missing"` relation would vary depending whether `resultTraitSet` was given.

I'm OK with assuming a well-formed library, and treating this missing entity case as "undefined behaviour", but if so I think this entry should be removed from the test library.

Reply:
I was being cheeky on the ACs, have done it properly now 🙃, added a proper check in BAL that all relations exist at query time [here](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/18/commits/70d8fa617c60438b674979e1ca7a38cf7894d5ce). I'll split out the exception base change once we're happy.

NB. I didn't wrap this as an OpenAssetIO exception as it is "exceptional" from an end-user perspective, and I think we generally accept there may be other exception types raise from arbitrary implementation code in exceptional situations.